### PR TITLE
HIVE-28604: Allow HMS to configure the DataNucleus level 1 cache

### PR DESCRIPTION
**Pull request by dengzhhu653**
### What changes were proposed in this pull request?
Allow HMS to configure the DataNucleus level 1 cache

### Why are the changes needed?
Under heavy memory load, a null value could be obtained from this cache,  leading to the opaque NPE.

### Does this PR introduce _any_ user-facing change?
Introduce a new property

### Is the change a dependency upgrade?
No

### How was this patch tested?
Test against a heavy HMS

### CODE REVIEW

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java [hunk 1]
```diff
@@ -724,6 +724,9 @@ public enum ConfVars {
     DATANUCLEUS_AUTOSTART("datanucleus.autoStartMechanismMode",
         "datanucleus.autoStartMechanismMode", "ignored", new StringSetValidator("ignored"),
         "Autostart mechanism for datanucleus.  Currently ignored is the only option supported."),
+    DATANUCLEUS_CACHE_LEVEL1_TYPE("datanucleus.cache.level1.type",
+        "datanucleus.cache.level1.type", "soft", new StringSetValidator("none", "soft", "weak", "strong"),
+        "Stores instances within a PersistenceManager"),
```
Comment (soumyakanti3578) [inline standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java:729]: Do we need a similar change for `DATANUCLEUS_CACHE_LEVEL2_TYPE` ?

**Review by soumyakanti3578 — COMMENTED** [review]
@dengzhhu653 could you please add a brief description about the tests that were performed to validate this, either here or preferably in the Jira?

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java (hunk 1 above)
Comment (dengzhhu653) [reply standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java:729]: it's already been here: https://github.com/apache/hive/blob/master/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java#L729

Comment (dengzhhu653) [reply standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java]: I would like to leave it as it is, there are multiple implementations for this level cache, some are DataNucleus-provided, others are external cache, https://www.datanucleus.org/products/accessplatform_4_2/extensions/level2_cache.html

**Review by saihemanth-cloudera — APPROVED** [review]
LGTM +1

**Review by soumyakanti3578 — APPROVED** [review]
LGTM

**Pull request merged**